### PR TITLE
more info in pc template & cmake compatibility

### PR DESCRIPTION
**Pull request by nilsnolde**
related to https://github.com/valhalla/valhalla/pull/4507

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -128,9 +128,15 @@ install(TARGETS prime_echod
 	prime_proxyd
 	prime_workerd DESTINATION ${CMAKE_INSTALL_BINDIR})
 
-CONFIGURE_FILE(
-	"${CMAKE_CURRENT_SOURCE_DIR}/libprime_server.pc.in"
-	"${CMAKE_CURRENT_BINARY_DIR}/libprime_server.pc"
+# pkg-config stuff, also make it compatible with autotools
+set(prefix ${CMAKE_INSTALL_PREFIX})
```
Comment (nilsnolde) [inline CMakeLists.txt:132]: the paths weren't set up in a cmake installation

File: CMakeLists.txt [hunk 2]
```diff
@@ -128,12 +129,22 @@ install(TARGETS prime_echod
...
+
+configure_file(
+	${CMAKE_CURRENT_SOURCE_DIR}/libprime_server.pc.in
+	${CMAKE_CURRENT_BINARY_DIR}/libprime_server.pc
 	@ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libprime_server.pc" DESTINATION ${pkgconfigdir})
 
+# uninstall target
+configure_file(cmake/uninstall.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/prime_server_uninstall.cmake @ONLY)
+add_custom_target(uninstall COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/prime_server_uninstall.cmake)
```
Comment (nilsnolde) [inline CMakeLists.txt:146]: same as in the valhalla PR https://github.com/valhalla/valhalla/pull/4516

File: libprime_server.pc.in [hunk 3]
```diff
@@ -7,4 +7,6 @@ Name: libprime_server
 Description: prime_server c++ library
 Version: @VERSION@
 Libs: -L${libdir} -lprime_server
+Requires: libzmq
+Requires.private: libczmq
```
Comment (kevinkreiser) [inline libprime_server.pc.in:11]: also curl

Comment (kevinkreiser) [inline libprime_server.pc.in:11]:
```suggestion
Requires.private: libczmq libcurl
```

**Review by kevinkreiser — APPROVED** [review]
looks good but had one suggestion

**Comment by kevinkreiser** [comment]
ill have to fix the stupid mac build separately..

**Pull request merged**